File: backend/app/api/v1/domain_checkins.py
"""
Domain Check-ins API — weekly explicit life domain ratings.

Framework alignment (March 2026): 7 life dimensions.

Endpoints:
- GET  /api/v1/domain-checkins/status  — check if a domain check-in is due
- POST /api/v1/domain-checkins         — submit 7 domain ratings
- GET  /api/v1/domain-checkins/history — get past check-in history
"""
from __future__ import annotations

from typing import List, Optional

from fastapi import Depends, Query
from pydantic import BaseModel, Field
from sqlalchemy.orm import Session

from app.core.database import get_db
from app.api.auth_mode import get_request_user_id
from app.api.router_factory import make_v1_router
from app.engine.domain_checkin_service import (
    get_domain_checkin_status,
    save_domain_checkin,
    get_domain_checkin_history,
)

router = make_v1_router(prefix="/api/v1/domain-checkins", tags=["domain-checkins"])


# ── Schemas (7 life dimensions) ──────────────────────────────────

class DomainCheckinRequest(BaseModel):
    session_id: Optional[int] = None
    career: float = Field(..., ge=1.0, le=10.0)
    relationship: float = Field(..., ge=1.0, le=10.0)
    family: float = Field(..., ge=1.0, le=10.0)
    health: float = Field(..., ge=1.0, le=10.0)
    finance: float = Field(..., ge=1.0, le=10.0)
    social: float = Field(..., ge=1.0, le=10.0)
    purpose: float = Field(..., ge=1.0, le=10.0)


class DomainCheckinStatusResponse(BaseModel):
    due: bool
    last_checkin_date: Optional[str] = None
    days_since: Optional[int] = None


class DomainCheckinResponse(BaseModel):
    id: int
    checkin_date: str
    career: float
    relationship: float
    family: float
    health: float
    finance: float
    social: float
    purpose: float


# ── Endpoints ────────────────────────────────────────────────────

@router.get("/status", response_model=DomainCheckinStatusResponse)
def check_status(
    user_id: int = Depends(get_request_user_id),
    db: Session = Depends(get_db),
):
    """Check whether a weekly domain check-in is due."""
    result = get_domain_checkin_status(db, user_id)
    return DomainCheckinStatusResponse(**result)


@router.post("", response_model=DomainCheckinResponse)
def submit_checkin(
    body: DomainCheckinRequest,
    user_id: int = Depends(get_request_user_id),
    db: Session = Depends(get_db),
):
    """Submit a weekly domain check-in with 7 domain ratings."""
    scores = {
        "career": body.career,
        "relationship": body.relationship,
        "family": body.family,
        "health": body.health,
        "finance": body.finance,
        "social": body.social,
        "purpose": body.purpose,
    }
    checkin = save_domain_checkin(db, user_id, body.session_id, scores)
    db.commit()

    return DomainCheckinResponse(
        id=checkin.id,
        checkin_date=checkin.checkin_date,
        career=checkin.career,
        relationship=checkin.relationship,
        family=checkin.family,
        health=checkin.health,
        finance=checkin.finance,
        social=checkin.social,
        purpose=checkin.purpose,
    )


@router.get("/history", response_model=List[DomainCheckinResponse])
def checkin_history(
    weeks: int = Query(default=12, ge=1, le=52),
    user_id: int = Depends(get_request_user_id),
    db: Session = Depends(get_db),
):
    """Get domain check-in history."""
    return get_domain_checkin_history(db, user_id, weeks=weeks)

File: frontend/src/api/domainCheckins.ts
/**
 * Domain Check-ins API client.
 */
import apiClient from './client';

export interface DomainCheckinStatus {
  due: boolean;
  last_checkin_date: string | null;
  days_since: number | null;
}

export interface DomainCheckinSubmit {
  session_id?: number | null;
  career: number;
  relationship: number;
  social: number;
  health: number;
  finance: number;
}

export interface DomainCheckinResponse {
  id: number;
  checkin_date: string;
  career: number;
  relationship: number;
  social: number;
  health: number;
  finance: number;
}

/**
 * Check if a weekly domain check-in is due.
 */
export async function getDomainCheckinStatus(): Promise<DomainCheckinStatus> {
  const res = await apiClient.get('/domain-checkins/status');
  return res.data;
}

/**
 * Submit a weekly domain check-in.
 */
export async function submitDomainCheckin(
  data: DomainCheckinSubmit,
): Promise<DomainCheckinResponse> {
  const res = await apiClient.post('/domain-checkins', data);
  return res.data;
}

/**
 * Get domain check-in history.
 */
export async function getDomainCheckinHistory(
  weeks: number = 12,
): Promise<DomainCheckinResponse[]> {
  const res = await apiClient.get('/domain-checkins/history', { params: { weeks } });
  return res.data;
}
